AirCasting/Extensions/Array+byRemovingDuplicates.swift

```diff
@@ -0,0 +1,50 @@
+// Created by Lunar on 30/08/2022.
+//
+import Foundation
+
+extension Numeric {
+    func decrement() -> Self {
+        self - 1
+    }
+}
+
+extension Array where Element: Equatable & Numeric {
+    /// Iterates all elements and gets rid of the duplicates using provided strategy
+    /// - Parameter strategy: a type of strategy which will be used inside `implement` function.
+    /// If this func returns `nil`, the new element won't be added to the resulting array.
+    /// If it returns a value, this value will be added to the resulting array.
+    /// - Returns: A new `Array` with conflicts resolved.
+    
+    enum StrategyType {
```

Comment:
I don't think changing the block-based strategy to `enum` is good idea.
If we ever want to change how duplicates are being dealt with in the future (which I can definitely see happening) we're gonna need to go to this file and implement new strategy here in this file.

For example if we ever want to get rid of duplicated streams in a stream array using sensor names we're gonna need to go here, add a new case to this `enum` and append `switch` statement below with the specific streams implementation. It will get messy and coupled.
With block-based strategy we provide a general solution that can be specialized onsite wherever we need it, however we need it, without any modification to this base implementation.
Consider this extension and overall this kind of extensions as a low-level building blocks that provide basic, universal behavior. It can be reused in many places or even across projects without ever changing the implementation. It's like `map` or `filter`.

(also this current approach narrows down the extension to *only* `Numerics` which is a bummer)

I also don't think this makes the callsite particularly easier to read, comparison:
`newValues.byRemovingDuplicates(strategy: .decrementation)`
`newValues.byRemovingDuplicates(strategy: { $0.decrement() })`

Reply:
Got it, thanks! I will change then